### PR TITLE
Call .native on ffi Fns; Use toArrayBuffer rather than toBuffer

### PR DESCRIPTION
**Pull request by cryptodeal**
Also minor correction to TensorInterface type

### CODE REVIEW

**Comment by cryptodeal** [comment]
Haven't previously contributed to a Facebook open source project, but just signed the CLA; automated check should pass shortly!

**Comment by bwasti** [comment]
Looks great!

There's a spot in the generated code that might benefit from the change to `.native`: https://github.com/facebookresearch/shumai/blob/main/scripts/gen_binding.py#L365
`fl._{op}` -> `fl._{op}.native`

you can then test this out by running `./scripts/gen_all_bindings.sh`

**Comment by cryptodeal** [comment]
Thanks for the feedback; I've updated gen_binding.py so that rather generating code using `fl._{op}`, the generated code uses `fl._{op}.native`.

**Comment by bwasti** [comment]
as discussed on discord, we'll need to fix up the coercion logic a bit: https://github.com/bwasti/shumai/commit/7c5d0a8f2477ad59722ea6c067de77e1f59d3fae

once that's done, we get a noticeable speedup!

before:
Screen Shot 2022-08-27 at 12 12 47 PM
after:
Screen Shot 2022-08-27 at 12 13 35 PM

**Comment by cryptodeal** [comment]
Added the change to gen_binding.py per your feedback on discord!

File: examples/bench.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import {
   Tensor,
   rand
-} from "shumaiml";
+} from "../shumai";
```
Comment (bwasti) [inline examples/bench.ts:4]: needs to be "shumaiml" again :)

File: src/tensor/tensor.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
 import {
   ptr,
   toArrayBuffer,
-  toBuffer,
-  FFIType,
-  ArrayBuffer
```
Comment (bwasti) [inline src/tensor/tensor.ts:6]: I think you need the ArrayBuffer type, still

**Review by bwasti — APPROVED** [review]
small nits, and then I can merge!

File: examples/bench.ts (hunk 1 above)
Comment (cryptodeal) [reply examples/bench.ts:4]: Oops, forgot to revert after testing; thanks!

**Comment by cryptodeal** [comment]
Both issues fixed; thanks for the feedback!

**Pull request merged**